### ISSUE TITLE
embedPng stuck in infinity loop

### ISSUE THREAD

**Issue by lukasa1993** [Hopding/pdf-lib#1146]
### What were you trying to do?

to embed png into pdf

### How did you attempt to do it?

https://github.com/lukasa1993/pdf-lib-issue

### What actually happened?

process stuck in loop

### What did you expect to happen?

to embed image into pdf

### How can we reproduce the issue?

here is the full POC where you can run working sample and infinite loop sample

so you can run `node test2.js` for working as expected sample and `node test.js` for looped one

https://github.com/lukasa1993/pdf-lib-issue

### Version

1.17.1

### What environment are you running pdf-lib in?

Node

### Checklist

- [X] My report includes a Short, Self Contained, Correct (Compilable) Example.
- [X] I have attached all PDFs, images, and other files needed to run my SSCCE.

### Additional Notes

so problem was that image PNG was not properly uploaded to server and then it was trying to embed it into pdf while MacOS preview still can show part of PNG pdf-lib just stuck in loop and hangs whole JS process cant even guard agains it with promise race condition i couldn't figure out any workaround

**Comment by dschi** [comment]
Did you ever find a way around this?

**Comment by lukasa1993** [comment]
no, i just try catch it and embed as jpg if it throws if both throw i just abandon   that image
